R/interface.R

```diff
@@ -45,6 +249,36 @@ dust <- function(alloc, run, reset, state, step, reorder, classname = "dust") {
 }
 
 
+dust_guess_type <- function(txt) {
+  re <- "^\\s*class\\s+([^{ ]+)\\s*(\\{.*|$)"
+  i <- grep(re, txt)
+  if (length(i) != 1L) {
+    stop("Could not automatically detect class name")
+  }
+  sub(re, "\\1", txt[[i]])
+}
+
+
+dust_workdir <- function(path) {
+  if (is.null(path)) {
+    path <- tempfile()
+  } else if (file.exists(path)) {
+    if (!is_directory(path)) {
+      stop(sprintf("Path '%s' already exists but is not a directory",
+                   path))
+    }
+    contents <- dir(path, recursive = TRUE, no.. = TRUE, include.dirs = FALSE)
```

Comment:
Is this the right setting for recursive? I just launched this with a close to top level directory and this caused it to hang

Reply:
Good point - will think about a better way of doing this